community/front-end/ofe/website/ghpcfe/models.py

```diff
@@ -534,6 +534,122 @@ def mount_source(self):
+
+    def __str__(self):
+        return self.name
+
+    name = models.CharField(
+        max_length=30,
+        help_text="Enter a startup script name",
+    )
+    description = models.TextField(
+        max_length=4000,
+        help_text="(Optional) description of this stratup script",
+        blank=True,
+        null=True,
+    )
+    STARTUP_SCRIPT_TYPES = (
+        ("shell", "Shell script"),
+        ("ansible-local", "Ansible playbook"),
+    )
+    type = models.CharField(
+        max_length=13,
+        choices=STARTUP_SCRIPT_TYPES,
+        blank=False,
+        default="shell",
+        help_text="Type of this application installation",
+    )
+    content = models.FileField(
+        upload_to="startup-scripts/",
+        help_text="Startup script file."
+    )
+    owner = models.ForeignKey(
+        User,
+        related_name="startup_script_owner",
+        help_text="Who owns this startup script?",
+        on_delete=models.RESTRICT,
+    )
+    authorised_users = models.ManyToManyField(
+        User,
+        related_name="startup_script_authorised_users",
+        help_text="Select other users authorised to use this startup script",
+    )
+        
+class Image(CloudResource):
+    """Model representing a custom node image."""
+
+    name = models.CharField(
+        max_length=30,
+        help_text="Enter an image name",
+        unique=True,
+    )
+
+    family = models.CharField(
+        max_length=30,
+        help_text="Enter you new image family",
+        unique=True,
+    )
+
+    source_image_project = models.CharField(
+        max_length=60,
+        help_text="Enter a source image project",
+        blank=False,
+        default="schedmd-slurm-public",
+    )
+
+    source_image_family = models.CharField(
+        max_length=60,
+        help_text="Enter a soure image family",
+        blank=False,
+        default="schedmd-v5-slurm-22-05-8-rocky-linux-8",
+    )
+
+    startup_script = models.ManyToManyField(
+        StartupScript,
+        help_text="Which startup scripts to use?",
+    )
+
+    enable_os_login = models.CharField(
+        max_length=5,
+        help_text="Enable OS Login durring the image creation?",
+        choices=(("TRUE", "TRUE"),("FALSE", "FALSE")),
+        default="TRUE",
+    )
+
+    block_project_ssh_keys = models.CharField(
+        max_length=5,
+        help_text="Don't use SSH keys in project metadata to create users?",
+        choices=(("TRUE", "TRUE"),("FALSE", "FALSE")),
+        default="TRUE",
+    )
+    owner = models.ForeignKey(
+        User,
+        related_name="image_owner",
+        help_text="Who owns this image?",
+        on_delete=models.RESTRICT,
+    )
+    authorised_users = models.ManyToManyField(
+        User,
+        related_name="image_authorised_users",
+        help_text="Select other users authorised to use this image",
+    )
+    IMAGE_STATUS = (
+        ("n", "Image is being newly configured by user"),
```

Comment:
May be best to have more descriptive keys such as "newly_configured", "being_created", "ready", "failed"